paraatm/simulation_method/aviationr.py

```diff
@@ -0,0 +1,210 @@
+        accident = False
+        accident_phase = case.loc[0, 'Phase_of_Flight'] # the initial phase of accident
+        while not accident:
+            self.simulationInterface.resume(delay)
+            while True:
+                runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                    break
+            aclist = self.aircraftInterface.getAllAircraftId()
+            ac = self.aircraftInterface.select_aircraft(aclist[0])
+            phase = ac.getFlight_phase()
+            if accident_phase in list(ntsb2nats.loc[ntsb2nats['NATS_CODE'] == phase]['NTSB_CODE']): # test if the phase of flight from  NASTS simulation is same as NTSB
+                accident = True
+
+        # Start accident simulation and risk analysis
+        risk_list = [None] * len(case)
+        subject_list = [None] * len(case)
+        phase_list = [None] * len(case)
+        occurrence_list = [None] * len(case)
+        time_list = [None] * len(case)
+        risk_estimator = RiskEstimator(self.model, isRNN, self.data, device)
+        model, hierarchical_softmax, risk_model = risk_estimator.load_model()
+        for i in range(1, len(case_code['Occurrence_Code'][0]) - 1):
+
+            # recording the risk and predicting events
+            risk_list[i - 1], phase_list[i - 1], occurrence_list[i - 1], subject_list[i - 1] = risk_estimator.risk_estimation(
+                case_code, i, device, model, hierarchical_softmax, risk_model, datadict, isRNN)
+
+            time_list[i-1] = self.simulationInterface.get_curr_sim_time()
+
+            # modify NATS simulation according to NTSB accident
+            t = self.accident_simulator(self.aircraftInterface, self.controllerInterface, self.pilotInterface, phase_list[i-1],
+                                        occurrence_list[i-1], subject_list[i-1], datadict, ntsb2nats)
+            self.simulationInterface.resume(t)
+            while True:
+                runtime_sim_status = self.simulationInterface.get_runtime_sim_status()
+                if (runtime_sim_status == self.NATS_SIMULATION_STATUS_PAUSE):
+                    break
+
+        self.simulationInterface.resume()
+        event_list = []
+        for i in range(len(subject_list)):
+            event_list.append(phase_list[i] + '\n' + occurrence_list[i] + '\n' + subject_list[i])
+        self.simulationInterface.write_trajectories(self.data + 'trajectory.csv')
+        return {'risk':risk_list, 'event':event_list, 'time':time_list}
+
+    def write_output(self, filename):
+        """
+        write NATS simulation output to specified file
+
+        Parameters
+        ----------
+        filename : str
+             directory to output file. Default in simulation function is out put in NATS_HOME with name
+             vcasmodule_xxxxxxxxxxxx.csv (xxxxxxxxxxxx is timestamp * 1000)
```

Comment:
Looks like this may be leftover from a copy/paste, and that it can be deleted.

Reply:
Got it. Thanks!